s3fs/core.py

```diff
@@ -1351,14 +1394,23 @@ async def _copy_managed(self, path1, path2, size, block=5 * 2 ** 30, **kwargs):
         )
         self.invalidate_cache(path2)
 
-    async def _cp_file(self, path1, path2, **kwargs):
-        gb5 = 5 * 2 ** 30
+    async def _cp_file(self, path1, path2, preserve_etag=None, **kwargs):
+        threshold = kwargs.pop("managed_threshold", 5 * 2 ** 30)
         path1 = self._strip_protocol(path1)
         bucket, key, vers = self.split_path(path1)
-        size = (await self._info(path1, bucket, key, version_id=vers))["size"]
-        if size <= gb5:
+
+        info = await self._info(path1, bucket, key, version_id=vers)
+        size = info["size"]
+        if size <= threshold:
             # simple copy allowed for <5GB
             await self._copy_basic(path1, path2, **kwargs)
+        elif preserve_etag:
+            # etag preserving multipart copy
+            _, _, parts_suffix = info["ETag"].strip('"').partition("-")
```

Comment:
It's new to me. Is there a URL explaining this, and do you think it might change in the future?

Reply:
A high level overview can be found here; https://teppen.io/2018/06/23/aws_s3_etags/.